datacube/drivers/postgis/_core.py

```diff
@@ -131,21 +141,22 @@ def ensure_db(engine, with_permissions: bool = True) -> bool:
             _LOG.info("Creating triggers.")
             install_timestamp_trigger(c)
             sqla_txn.commit()
-            if with_permissions:
-                c.execute(text(f"set role {quoted_user}"))
             c.commit()
             # Stamp with latest Alembic revision
             alembic_cfg = config.Config(ALEMBIC_INI_LOCATION)
             alembic_cfg.attributes["connection"] = c
             command.stamp(alembic_cfg, "head")
+            if with_permissions:
+                c.execute(text(f"set role {quoted_user}"))
+        # If not new, caller updates with alembic
```

Comment:
I had to think for several minutes to understand what this comment means.

I'm not sure how much useful information the comment contributes since I assume alembic is the default for SQLAlchemy, but if the comment should remain, I think it should be on the line before `if is_new:` instead of here.